src/lib/form-field/form-field.ts

```diff
@@ -336,7 +338,10 @@ export class MatFormField extends _MatFormFieldMixinBase
     });
 
     if (this._dir) {
-      this._dir.change.pipe(takeUntil(this._destroyed)).subscribe(() => this.updateOutlineGap());
+      this._dir.change.pipe(takeUntil(this._destroyed)).subscribe(() => {
+        this.updateOutlineGap();
+        this._previousDirection = this._dir.value;
```

Comment:
why is the direction not updated until after calling `updateOutlineGap`?

Reply:
Oh sorry I just read your explanation above, can you add a comment to the code. It looks wrong at first glance, without a note explaining it.

Reply:
Even looking at the description on the PR, I'm not quite sure whether this will cover all the cases. [The event will only if a new value is set](https://github.com/angular/material2/blob/master/src/cdk/bidi/dir.ts#L56). Also we probably need a unit test for this.

Reply:
@crisbeto @mmalerba "gulp test" and "gulp serve:devapp" are not working in my local. Do I need to do any configuration changes? Can you help me in running test cases? Even I tried "karma run", that's also giving errors in the console.

Reply:
gulp test is working fine when i updated the node packages. But is there any way to run specific test suite ?

Reply:
@crisbeto @mmalerba I added test cases and comments. Can you please review those.